### DISCUSSION TITLE
Mouse Scroll not working on MacOS

### DISCUSSION THREAD

**Discussion by msibal6** [ocornut/imgui#5664] [New Users Build/Link/Run issues ONLY!]
MacOS: 10.14
imgui: 1.79. 

In my imgui opengl3 example, I am able to interact with every part to the gui except the mouse scroll wheel delta does not show any changes. In my own little test, I was able to see that I get scroll input in my glfw scroll back, so I know the application can see. I'm just confused why imgui cannot detect it. Any help would be appreciated. Thank you! 

msibal6

**Comment by PathogenDavid** [comment]
> I get scroll input in my glfw scroll back

Make sure that you either:

A) Install your callback before calling `ImGui_ImplGlfw_InitForOpenGL`
B) Properly store and chain-call the return value from `glfwSetScrollCallback`
C) Manually call `ImGui_ImplGlfw_ScrollCallback` form your callback

(As an aside, discussions are not the correct place for this sort of question. Next time please [create an issue](https://github.com/ocornut/imgui/issues/new))

**Reply by msibal6** [reply]
Thanks for the help! Manually calling `ImGui_ImplGlfw_Scrollback` from the callback but I just realized that none of the key downs are working in my application. I'll have to look at the example  a little bug so I can fix this bug. I'll be sure to make an issue next time I still have the problem 😄
